packages/ui-text-area/src/TextArea/index.tsx

```diff
@@ -98,142 +104,113 @@ class TextArea extends Component<TextAreaProps> {
 
-      // @ts-expect-error ts-migrate(2339) FIXME: Property '_textarea' does not exist on type 'TextA... Remove this comment to see the full error message
       if (this._textarea && !this._textareaResizeListener) {
-        // @ts-expect-error ts-migrate(2339) FIXME: Property 'height' does not exist on type '{}'.
         const { height: origHeight } = getBoundingClientRect(this._textarea)
-        // @ts-expect-error ts-migrate(2339) FIXME: Property '_textareaResizeListener' does not exist ... Remove this comment to see the full error message
         this._textareaResizeListener = new ResizeObserver((entries) => {
           for (const entry of entries) {
             const { height } = entry.contentRect
 
             if (origHeight !== height) {
-              // @ts-expect-error ts-migrate(2554) FIXME: Expected 1 arguments, but got 0.
               this._textareaResize()
             }
           }
         })
 
-        // @ts-expect-error ts-migrate(2339) FIXME: Property '_textareaResizeListener' does not exist ... Remove this comment to see the full error message
         this._textareaResizeListener.observe(this._textarea)
       }
 
       this._request = requestAnimationFrame(this.grow)
     }
   }
 
-  // @ts-expect-error ts-migrate(6133) FIXME: 'evt' is declared but its value is never read.
-  grow = (evt) => {
-    // @ts-expect-error ts-migrate(2339) FIXME: Property '_textarea' does not exist on type 'TextA... Remove this comment to see the full error message
+  grow = () => {
     if (!this._textarea || this._manuallyResized) {
       return
     }
-    // @ts-expect-error ts-migrate(2339) FIXME: Property '_textarea' does not exist on type 'TextA... Remove this comment to see the full error message
     const offset = this._textarea.offsetHeight - this._textarea.clientHeight
     let height = ''
 
     // Notes:
     // 1. height has to be reset to `auto` every time this method runs, or scrollHeight will not reset
     // 2. `this._textarea.scrollHeight` will not reset if assigned to a variable; it needs to be written out each time
-    // @ts-expect-error ts-migrate(2339) FIXME: Property '_textarea' does not exist on type 'TextA... Remove this comment to see the full error message
     this._textarea.style.height = 'auto'
-    // @ts-expect-error ts-migrate(2339) FIXME: Property '_textarea' does not exist on type 'TextA... Remove this comment to see the full error message
     this._textarea.style.overflowY = 'hidden' // hide scrollbars for autoGrow textareas
-    // @ts-expect-error ts-migrate(2339) FIXME: Property '_textarea' does not exist on type 'TextA... Remove this comment to see the full error message
     height = this._textarea.scrollHeight + offset + 'px'
 
-    // @ts-expect-error ts-migrate(2339) FIXME: Property '_container' does not exist on type 'Text... Remove this comment to see the full error message
-    const maxHeight = px(this.props.maxHeight, this._container)
+    const maxHeight = this.props.maxHeight
```

Comment:
Looks good, but how about just wrapping this block in an `if`? Seems a bit more readable, and it shows the block is only relevant if the user provided a `maxHeight` prop :)

```js
  if (this.props.maxHeight) {
      const maxHeight = px(this.props.maxHeight, this._container)

      if (this._textarea.scrollHeight > maxHeight) {
        this._textarea.style.overflowY = 'auto' // add scroll if scrollHeight exceeds maxHeight in pixels
      } else if (this.props.height) {
        if (this._textarea.value === '') {
          height = this.props.height
        } else if (
          px(this.props.height, this._container) > this._textarea.scrollHeight
        ) {
          this._textarea.style.overflowY = 'auto' // add scroll if scrollHeight exceeds height in pixels
          height = this.props.height
        }
      }

      // preserve container height to prevent scroll jumping on long textareas,
      // but make sure container doesn't exceed maxHeight prop
      const heightExceedsMax = px(height) > maxHeight
      if (!heightExceedsMax) {
        this._container!.style.minHeight = height
      }
    }
```

Reply:
When this.props.maxHeight `falsy`, maxHeight is also `falsy`, do heightExceedsMax is `falsy`, so the
` if (!heightExceedsMax) {
        this._container!.style.minHeight = height
      }`
should run, which will not if we put the `if` there